sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestProxyDownloader.java

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.utils;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.logging.LogLevel;
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveException;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.ArchiveStreamFactory;
+import org.apache.commons.compress.utils.IOUtils;
```

Comment:
Do we need the Apache compression utils? Could the JDK compression API work?

Reply:
Based on some reading I did, the JDK compression API doesn't nicely support `.tar.gz`. Apache does. (Though in this case I am not certain what I implemented here will work for `.tar.gz` - but the APIs exist to make it work.)